lib/vty.c

```diff
@@ -2137,11 +2137,12 @@ void vty_close(struct vty *vty)
 	/* Close socket. */
 	if (vty->fd > 0) {
 		close(vty->fd);
-		if (vty->wfd > 0 && vty->wfd != vty->fd)
-			close(vty->wfd);
 	} else
 		was_stdio = true;
 
+	if (vty->wfd > 0 && vty->wfd != vty->fd)
+		close(vty->wfd);
```

Comment:
This is not correct; the wfd is intentionally left open if it is stdout because the logging code can use it even after the stdio-vty is closed.